code/iaas/config-item/common/src/main/java/io/cattle/platform/configitem/model/Client.java

```diff
@@ -34,27 +38,37 @@ public Client(ConfigItemStatus status) {
         } else if (status.getEnvironmentId() != null) {
             resourceType = Environment.class;
             resourceId = status.getEnvironmentId();
+        } else if (status.getHostId() != null) {
+            resourceType = Host.class;
+            resourceId = status.getHostId();
         }
 
-        assignEvent();
+        assignEvent(status.getName());
     }
 
     public Client(Class<?> resourceType, long resourceId) {
         super();
         this.resourceType = resourceType;
         this.resourceId = resourceId;
-        assignEvent();
+        assignEvent(null);
     }
 
-    protected void assignEvent() {
+    protected void assignEvent(String configItemStatusName) {
         if (resourceType == Agent.class) {
             eventName = IaasEvents.CONFIG_UPDATE;
         } else if (resourceType == Account.class) {
             eventName = IaasEvents.GLOBAL_SERVICE_UPDATE;
         } else if (resourceType == Service.class) {
-            eventName = IaasEvents.SERVICE_UPDATE;
+
+            if (StringUtils.equals(SERVICE_ENDPOINTS_UPDATE_ITEM_NAME, configItemStatusName)) {
```

Comment:
@ibuildthecloud  we map the db field to iaas event today, may be we should assign an event on per config item name instead?